libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -200,7 +204,7 @@ method publish*(p: PubSub,
 
 method initPubSub*(p: PubSub) {.base.} =
   ## perform pubsub initializaion
-  discard
+  p.observers = new(seq[PubSubObserver])
```

Comment:
Why `PubSubPeer` needs this observers list?

Reply:
Intercept `encode`